tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+
+        with self.assertRaises(pexEx.DomainError):
+            northPole.bearingTo(safe)
+        with self.assertRaises(pexEx.DomainError):
+            southPole.bearingTo(safe)
+
+    def testBearingToValue(self):
+        """Test if bearingTo() returns the expected value.
+        """
+        lon0 = 90.0
+        lat0 = 0.0   # These tests only work from the equator.
+        arcLen = 10.0
+
+        trials = [
+            # Along celestial equator
+            dict(lon=lon0, lat=lat0, bearing=0.0, lonEnd=lon0+arcLen, latEnd=lat0),
+            # Along a meridian
+            dict(lon=lon0, lat=lat0, bearing=90.0, lonEnd=lon0, latEnd=lat0+arcLen),
+            # 180 degree arc (should go to antipodal point)
+            dict(lon=lon0, lat=lat0, bearing=45.0, lonEnd=lon0+180.0, latEnd=-lat0),
+            #
+            dict(lon=lon0, lat=lat0, bearing=45.0, lonEnd=lon0+90.0, latEnd=lat0+45.0),
+            dict(lon=lon0, lat=lat0, bearing=225.0, lonEnd=lon0-90.0, latEnd=lat0-45.0),
+            # Argument at a pole should work
+            dict(lon=lon0, lat=lat0, bearing=270.0, lonEnd=lon0, latEnd=-90.0),
+            # Support for non-finite values
+            dict(lon=lon0, lat=nan, bearing=nan, lonEnd=lon0, latEnd=45.0),
+            dict(lon=lon0, lat=lat0, bearing=nan, lonEnd=nan, latEnd=90.0),
+            dict(lon=inf, lat=lat0, bearing=nan, lonEnd=lon0, latEnd=42.0),
+            dict(lon=lon0, lat=lat0, bearing=nan, lonEnd=-inf, latEnd=42.0),
+        ]
+
+        for trial in trials:
+            origin = SpherePoint(trial['lon']*degrees, trial['lat']*degrees)
+            end = SpherePoint(trial['lonEnd']*degrees, trial['latEnd']*degrees)
+            bearing = origin.bearingTo(end)
+
+            self.assertIsInstance(bearing, afwGeom.Angle)
+            if origin.isFinite() and end.isFinite():
+                self.assertGreaterEqual(bearing.asDegrees(), 0.0)
+                self.assertLess(bearing.asDegrees(), 360.0)
+            if origin.separation(end).asDegrees() != 180.0:
+                if not math.isnan(trial['bearing']):
+                    self.assertAlmostEqual(trial['bearing'], bearing.asDegrees(), 12)
+                else:
+                    self.assertTrue(math.isnan(bearing.asRadians()))
+
+    def testBearingToValueSingular(self):
+        """White-box test: bearingTo() may be unstable if points are near opposite poles.
+        """
+        southPole = SpherePoint(0.0*degrees, (-90.0 + 1e-14)*degrees)
+        northPoleSame = SpherePoint(0.0*degrees, (90.0 - 1e-14)*degrees)
+        # Don't let it be on exactly the opposite side.
+        northPoleOpposite = SpherePoint(180.0*degrees, (90.0 - 1e-13)*degrees)
+
+        self.assertAnglesNearlyEqual(southPole.bearingTo(northPoleSame), afwGeom.HALFPI*afwGeom.radians)
+        self.assertAnglesNearlyEqual(southPole.bearingTo(northPoleOpposite),
+                                     (afwGeom.PI + afwGeom.HALFPI)*afwGeom.radians)
```

Comment:
Please also test from near each pole to some point that is not near a pole (perhaps on the equator or in another spot there the expected answer is obvious).

Similarly, please make such tests for `offset` and `rotated`. I think those are the only methods sensitive to numerical errors near the pole, though I may have missed something.